app/upgrades/v14/upgrades.go

```diff
@@ -16,6 +19,28 @@ func CreateUpgradeHandler(
 	keepers *keepers.AppKeepers,
 ) upgradetypes.UpgradeHandler {
 	return func(ctx sdk.Context, plan upgradetypes.Plan, fromVM module.VersionMap) (module.VersionMap, error) {
+		keepers.SwapRouterKeeper.SetParams(ctx, swaproutertypes.DefaultParams())
+
+		// N.B: pool id in gamm is to be deprecated in the future
+		// Instead,it is moved to swaprouter.
+		migrateNextPoolId(ctx, keepers.GAMMKeeper, keepers.SwapRouterKeeper)
+
+		//  N.B.: this is done to avoid initializing genesis for swaprouter module.
+		// Otherwise, it would overwrite migrations with InitGenesis().
+		// See RunMigrations() for details.
```

Comment:
Consider adding why setting fromVM to zero prevents the overwrite directly here

Reply:
Not sure what you mean - the note attempts to explain the why. This is to avoid initializing genesis for swaprouter module and prevent overwriting the logic of the migration we perform in the upgrade handler here.

If you look at the linked `RunMigrations`, it should make it clearer